junit-jupiter-params/src/main/java/org/junit/jupiter/params/ParameterizedTestParameterResolver.java

```diff
@@ -65,6 +72,42 @@ public Object resolveParameter(ParameterContext parameterContext, ExtensionConte
 		return this.methodContext.resolve(parameterContext, extractPayloads(this.arguments));
 	}
 
+	/**
+	 * @since 5.8
+	 */
+	@Override
+	public void afterTestExecution(ExtensionContext context) {
+		ParameterizedTest parameterizedTest = AnnotationUtils.findAnnotation(context.getRequiredTestMethod(),
+			ParameterizedTest.class).get();
+		if (!parameterizedTest.autoCloseArguments()) {
+			return;
+		}
+
+		Store store = context.getStore(NAMESPACE);
+		AtomicInteger argumentIndex = new AtomicInteger();
+
+		Arrays.stream(arguments) //
+				.filter(AutoCloseable.class::isInstance) //
+				.map(AutoCloseable.class::cast) //
+				.map(CloseableArgument::new) //
+				.forEach(closeable -> store.put("closeableArgument" + argumentIndex.getAndIncrement(), closeable));
```

Comment:
That turned out _nice & concise_. 👍

Reply:
Yay, thanks again for the tip! :ok_hand: